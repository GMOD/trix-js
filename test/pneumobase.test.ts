import { LocalFile } from 'generic-filehandle2'
import { expect, test } from 'vitest'

import Trix from '../src/index'

test('can find pneumobase features', async () => {
  const trix1 = new Trix(
    new LocalFile(
      './test/testData/D39V_annotation_coding_features_sorted.gff.ixx',
    ),
    new LocalFile(
      './test/testData/D39V_annotation_coding_features_sorted.gff.ix',
    ),
  )
  const hitList = await trix1.search('SPV_05')
  expect(hitList).toMatchSnapshot()
})
