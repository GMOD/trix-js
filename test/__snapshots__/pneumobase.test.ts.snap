// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`can find pneumobase features 1`] = `
[
  [
    "spv_0500",
    "["D39V%3A505863..508558"|"D39V_annotation_coding_features_sorted.gff"|"SPV_0500"|"SPD_0500"|"SPV_0500"|"spr0500"|"SP_0574%20%20SP_0575"|"SPV_RS11315"]",
  ],
  [
    "spv_0501",
    "["D39V%3A508859..509698"|"D39V_annotation_coding_features_sorted.gff"|"bglG"|"SPD_0501"|"SPV_0501"|"spr0504"|"SP_0576"|"SPV_RS02690"|"licT"]",
  ],
  [
    "spv_0502",
    "["D39V%3A509716..511554"|"D39V_annotation_coding_features_sorted.gff"|"bglF"|"SPD_0502"|"SPV_0502"|"spr0505"|"SP_0577"|"SPV_RS02695"]",
  ],
  [
    "spv_0503",
    "["D39V%3A511567..512982"|"D39V_annotation_coding_features_sorted.gff"|"bglA-2"|"SPD_0503"|"SPV_0503"|"spr0506"|"SP_0578"|"SPV_RS02700"|"bglB%20%20bglH"]",
  ],
  [
    "spv_0504",
    "["D39V%3A513574..514620"|"D39V_annotation_coding_features_sorted.gff"|"pheS"|"SPD_0504"|"SPV_0504"|"spr0507"|"SP_0579"|"SPV_RS02705"]",
  ],
  [
    "spv_0505",
    "["D39V%3A514620..515129"|"D39V_annotation_coding_features_sorted.gff"|"paiA"|"SPD_0505"|"SPV_0505"|"spr0508"|"SP_0580"|"SPV_RS02710"]",
  ],
  [
    "spv_0506",
    "["D39V%3A515206..517608"|"D39V_annotation_coding_features_sorted.gff"|"pheT"|"SPD_0506"|"SPV_0506"|"spr0509"|"SP_0581"|"SPV_RS02715"]",
  ],
  [
    "spv_0507",
    "["D39V%3A517676..518677"|"D39V_annotation_coding_features_sorted.gff"|"SPV_0507"|"SPD_0507"|"SPV_0507"|"spr0510"|"SP_0582"|"SPV_RS02720"]",
  ],
  [
    "spv_0508",
    "["D39V%3A518833..519322"|"D39V_annotation_coding_features_sorted.gff"|"SPV_0508"|"SPD_0508"|"SPV_0508"|"spr0511"|"SP_0583"|"SPV_RS02725"]",
  ],
  [
    "spv_0509",
    "["D39V%3A519406..519672"|"D39V_annotation_coding_features_sorted.gff"|"higA"|"SPD_0509"|"SPV_0509"|"spr0513"|"SP_0584"|"SPV_RS02730"]",
  ],
  [
    "spv_0510",
    "["D39V%3A519933..522182"|"D39V_annotation_coding_features_sorted.gff"|"metE"|"SPD_0510"|"SPV_0510"|"spr0514"|"SP_0585"|"SPV_RS02735"]",
  ],
  [
    "spv_0511",
    "["D39V%3A522246..523112"|"D39V_annotation_coding_features_sorted.gff"|"metF"|"SPD_0511"|"SPV_0511"|"spr0515"|"SP_0586"|"SPV_RS02740"]",
  ],
  [
    "spv_0512",
    "["D39V%3A523797..526010"|"D39V_annotation_coding_features_sorted.gff"|"pnp"|"SPD_0512"|"SPV_0512"|"spr0516"|"SP_0588"|"SPV_RS02750"|"pnpA"]",
  ],
  [
    "spv_0513",
    "["D39V%3A526026..526643"|"D39V_annotation_coding_features_sorted.gff"|"cysE"|"SPD_0513"|"SPV_0513"|"spr0517"|"SP_0589"|"SPV_RS02755"]",
  ],
  [
    "spv_0514",
    "["D39V%3A526655..527539"|"D39V_annotation_coding_features_sorted.gff"|"SPV_0514"|"SPD_0514"|"SPV_0514"|"spr0518"|"SP_0590"|"SPV_RS02760"]",
  ],
  [
    "spv_0515",
    "["D39V%3A527621..528964"|"D39V_annotation_coding_features_sorted.gff"|"cysS"|"SPD_0515"|"SPV_0515"|"spr0519"|"SP_0591"|"SPV_RS02770"]",
  ],
  [
    "spv_0516",
    "["D39V%3A528957..529343"|"D39V_annotation_coding_features_sorted.gff"|"mrnC"|"SPD_0516"|"SPV_0516"|"spr0520"|"SP_0592"|"SPV_RS02775"]",
  ],
  [
    "spv_0517",
    "["D39V%3A529347..530231"|"D39V_annotation_coding_features_sorted.gff"|"SPV_0517"|"SPD_0517"|"SPV_0517"|"spr0521"|"SP_0593"|"SPV_RS02780"]",
  ],
  [
    "spv_0521",
    "["D39V%3A531637..532914"|"D39V_annotation_coding_features_sorted.gff"|"vex1"|"SPD_0521"|"SPV_0521"|"spr0524"|"SP_0599"|"SPV_RS02790"]",
  ],
  [
    "spv_0522",
    "["D39V%3A532927..533574"|"D39V_annotation_coding_features_sorted.gff"|"vex2"|"SPD_0522"|"SPV_0522"|"spr0525"|"SP_0600"|"SPV_RS02795"]",
  ],
]
`;
